SmartDeviceLink/SDLCacheFileManager.m

```diff
@@ -0,0 +1,282 @@
+        }
+        
+        // Save lock screen icon to file path
+        NSString *iconFilePath = [self sdl_writeImage:image toFileFromURL:request.url];
+
+        if (iconFilePath == nil) {
+            SDLLogE(@"Could not save lockscreen icon to path: %@", error);
+        }
+        
+        // Update archive file with icon
+        [self updateArchiveFileFromRequest:request iconFilePath:iconFilePath archiveFile:iconArchiveFile];
+        
+        return completion(image, nil);
+    }];
+}
+
+#pragma mark - Cache Helper Methods
+
+/**
+ * Handles request to retrieve lock screen archive file from specified path.
+ *
+ * @param path The directory where the archive file is held
+ * @return The archive file at path
+*/
+- (nullable SDLIconArchiveFile *)sdl_retrieveArchiveFileFromPath:(NSString *)path {
+    NSString *archiveObjectPath = self.sdl_archiveFileDirectory;
+    SDLIconArchiveFile *iconArchiveFile = [NSKeyedUnarchiver unarchiveObjectWithFile:archiveObjectPath];
+    
+    return iconArchiveFile;
+}
+
+/**
+ * Handles the request to clear all files from lock screen icon directory in cache.
+*/
+- (void)sdl_clearLockScreenDirectory {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSString *directory = self.sdl_cacheFileBaseDirectory;
+    NSError *error = nil;
+    
+    for (NSString *file in [fileManager contentsOfDirectoryAtPath:directory error:&error]) {
+        if (error != nil) {
+            SDLLogE(@"Failed retrieving contents of directory: %@ with error: %@", directory, error);
+        }
+        BOOL success = [fileManager removeItemAtPath:[directory stringByAppendingPathComponent:file] error:&error];
+        if (!success || error != nil) {
+            SDLLogE(@"Could not delete file: %@", error);
+            break;
+        }
+    }
+}
+
+/**
+ * Handles request to save icon image to cache.
+ *
+ * @param icon The icon to be saved to the cache
+ * @param urlString The url of the icon to be hashed and used as file name
+ * @return A string representation of the directory the icon was saved to
+*/
+- (nullable NSString *)sdl_writeImage:(UIImage *)icon toFileFromURL:(NSString *)urlString {
+    NSData *iconPNGData = UIImagePNGRepresentation(icon);
+    NSString *iconStorageName = [self.class sdl_md5HashFromString:urlString];
+    
+    NSString *imageFilePath = [self.sdl_cacheFileBaseDirectory stringByAppendingPathComponent:iconStorageName];
+    [iconPNGData writeToFile:imageFilePath atomically:YES];
+    
+    return imageFilePath;
+}
+
+/**
+ * Handles request to update archive file. Will create a new archive file if one is not present.
+ *
+ * @param request The System request used to retrieve the icon
+ * @param iconFilePath The directory path where the icon file is held
+ * @param archiveFile Current archive file to update
+*/
+- (void)updateArchiveFileFromRequest:(SDLOnSystemRequest *)request iconFilePath:(NSString *)iconFilePath archiveFile:(SDLIconArchiveFile *)archiveFile {
```

Comment:
```suggestion
- (void)updateArchiveFileWithIconURL:(NSString *)iconURL iconFilePath:(NSString *)iconFilePath archiveFile:(SDLIconArchiveFile *)archiveFile {
```

It seems like you only use the URL, so that's the only thing you have to pass